### ISSUE TITLE
Evaluating on unlabeled dataset reproduces arbitrary assigned labels

### ISSUE THREAD

**Issue by austinweigle** [tyang816/ProtSolM#5]
After confirming my ProtSolM installation works, I extracted features from my prepared PDBs. I am testing points to mutations on a single protein. So there may be a need for greater sensitivity when evaluating. I also made a `test.csv` file.

However, when I follow instructions for using a custom dataset with `eval.py`, I realize there is no way to reference the PDBSol database that you prepared. That is - the install and instructions are basically good for trying to recreate what you did in the published work, but with a separate or new database.

It does not appear there is a way to inference a model trained on the PDBSol data to actually make predictions on unlabeled proteins. If I put '0' or '1'  as arbitrary labels for my proteins, they simply are assigned the exact same labels after I run the model. I am not discrediting the work, but in terms of my current implementation, it does not appear that I can get much value from your software.

Is there something I'm doing wrong? I'm following the exact instructions from the README for "Your own dataset". I was thinking that maybe I would have to run my predictions along with some of the training data? Like a random split or bootstrapping? However, my local install does not have enough memory to reperform a benchmark of the entire PDBSol dataset.

**Comment by tyang816** [comment]
Thank you very much for using our model and for your feedback!

1. The ProtSolM training set does not include any mutation data — it is trained only on natural proteins. Therefore, it is not designed to predict the solubility of mutated proteins. To do that, you would need to retrain the model using a dataset that includes mutations.
2. The PDBSol dataset we used is available on GitHub at: data/PDBSol.
3. Could you please share your file or a screenshot of the output? In our prediction results using PDBSol, there are two columns: label and pred_label. The pred_label column shows the predicted values from the model.
4. I would really appreciate more detailed screenshots or a description of your files — it will help me improve the tool and provide better support.

Best regards,
Yang Tan
